xray/src/generation.rs

```diff
@@ -676,11 +617,100 @@ pub fn build_xray_quadtree(
     };
 
     let meta_pb_name = format!("{}.pb", root_node_id).replace("r", "meta");
-    let mut buf_writer = BufWriter::new(File::create(output_directory.join(meta_pb_name)).unwrap());
+    let mut buf_writer =
+        BufWriter::new(File::create(&parameters.output_directory.join(meta_pb_name)).unwrap());
     meta.write_to_writer(&mut buf_writer).unwrap();
 
-    progress_bar.lock().unwrap().finish();
+    Ok(())
+}
+
+pub fn create_leaf_nodes(
+    leaf_nodes: Vec<Node>,
+    deepest_level: u8,
+    bounding_box: &Aabb3<f64>,
+    coloring_strategy_kind: &ColoringStrategyKind,
+    parameters: &XrayParameters,
+) -> ImageResult<FnvHashSet<NodeId>> {
+    let (created_leaf_node_ids_tx, created_leaf_node_ids_rx) = crossbeam::channel::unbounded();
+    let progress_bar = create_syncable_progress_bar(
+        leaf_nodes.len(),
+        &format!("Building level {}", deepest_level),
+    );
+    leaf_nodes
+        .into_par_iter()
+        .try_for_each(|node| -> ImageResult<()> {
+            let strategy: Box<dyn ColoringStrategy> = coloring_strategy_kind.new_strategy();
+            let rect_min = node.bounding_rect.min();
+            let rect_max = node.bounding_rect.max();
+            let min = Point3::new(rect_min.x, rect_min.y, bounding_box.min().z);
+            let max = Point3::new(rect_max.x, rect_max.y, bounding_box.max().z);
+            let bbox = Aabb3::new(min, max);
+            if let Some(image) = xray_from_points(
+                &bbox,
+                Vector2::new(parameters.tile_size_px, parameters.tile_size_px),
+                strategy,
+                parameters,
+            ) {
+                image.save(&get_image_path(&parameters.output_directory, node.id))?;
+                created_leaf_node_ids_tx.send(node.id).unwrap();
+            }
+            progress_bar.lock().unwrap().inc();
+            Ok(())
+        })?;
+    progress_bar.lock().unwrap().finish_println("");
+    drop(created_leaf_node_ids_tx);
+    Ok(created_leaf_node_ids_rx.into_iter().collect())
+}
+
+pub fn create_non_leaf_nodes(
+    created_leaf_node_ids: FnvHashSet<NodeId>,
+    deepest_level: u8,
+    root_level: u8,
+    parameters: &XrayParameters,
```

Comment:
If instead of passing all `XrayParameters` we pass `output_directory`, `tile_size_px`, and `tile_background_color`, as separate arguments we could reuse this function in merger.

Reply:
Cool, done